Update many dependencies

dnewh opened the pull request:
This updates many dependencies that were reporting security
vulnerabilities.

Signed-off-by: Davey Newhall <newhall@bitwise.io>

dnewh committed:
Update deps with reported security vulnerabilities

This updates many dependencies that were reporting security
vulnerabilities.

Signed-off-by: Davey Newhall <newhall@bitwise.io>

peterschwarz reviewed (changes requested):
Change the commit summary to:

"Update deps with reported security vulnerabilities"

The current summary is too vague

peterschwarz commented on app/package.json:
```diff
@@ -34,20 +34,20 @@
   },
   "devDependencies": {
-    "eslint": "^6.6.0",
+    "eslint": "^7.24.0",
```
Doesn't this need to be fixed at `7.22.0`, due to a bug in the linter?

dnewh committed:
Pin eslint dependency

This pins eslint due to bug:
https://github.com/yannickcr/eslint-plugin-react/issues/2960

Signed-off-by: Davey Newhall <newhall@bitwise.io>

dnewh committed:
Use published Transact SDK

This updates the profile and circuits saplings to use the published
version of the Transact SDK. This previously was included with Github
deps.

Signed-off-by: Davey Newhall <newhall@bitwise.io>
